packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx

```diff
@@ -82,33 +86,43 @@ const EntityVersionsDetailsContainerImpl: React.FC<
   const id = workflowId as ResourceIdentifier;
   const sections = entitySections[id.resourceType];
   const versionsSections = versionsDetailsSections[id.resourceType];
-  const project = useProject(workflowId.project);
+  const [project] = useProject(workflowId.project);
   const styles = useStyles({ resourceType: id.resourceType });
 
   return (
-    <WaitForData {...project}>
-      <EntityDetailsHeader
-        project={project.value}
-        id={id}
-        launchable={sections.launch}
-        backToWorkflow
-      />
-      <div className={styles.verionDetailsContainer}>
-        {versionsSections.details && (
-          <div className={styles.versionDetailsContainer}>
-            <EntityVersionDetails id={id} />
+    <>
+      {project?.id ? (
+        <>
+          <Box pl={2} pr={2}>
+            <EntityDetailsHeader
+              project={project.value}
+              id={id}
+              launchable={sections.launch}
+              backToWorkflow
+            />
+          </Box>
+          <div className={styles.verionDetailsContainer}>
+            {versionsSections.details && (
+              <div className={styles.versionDetailsContainer}>
+                <EntityVersionDetails id={id} />
+              </div>
+            )}
+            {versionsSections.graph && (
+              <div className={styles.staticGraphContainer}>
+                <StaticGraphContainer workflowId={workflowId} />
+              </div>
+            )}
+            <div className={styles.versionsContainer}>
+              <EntityVersions id={id} showAll />
+            </div>
           </div>
-        )}
-        {versionsSections.graph && (
-          <div className={styles.staticGraphContainer}>
-            <StaticGraphContainer workflowId={workflowId} />
-          </div>
-        )}
-        <div className={styles.versionsContainer}>
-          <EntityVersions id={id} showAll />
-        </div>
-      </div>
-    </WaitForData>
+        </>
+      ) : (
```

Comment:
Please catch the loader state early via a return

if (!project.id) {
  return <LoadingSpinner />
}


...